### ISSUE TITLE
[F.Rq.] Allow both static and input-bound markers

### ISSUE THREAD

**Issue by RetroEdit** [TASEmulators/BizHawk#4072]
Bind markers to input is a clunky setting. Both frame-static markers and markers bound to inputs are useful. I find myself just keeping the checkbox on since I generally TAS sync-friendly games and use markers as a sort of "self-adjusting bookmarks" system that can account for insert/delete frames earlier in the movie. But markers can also be useful as a measuring stick to say "previously XYZ happened on *this* frame". The bind markers setting also [changes how branches work](https://github.com/TASEmulators/BizHawk/issues/3899#issuecomment-2067633501).

As for how this gets implemented in practice: I think a simple implementation is for each marker to have a MarkerType that appears as a checkbox underneath the marker name when you edit it. A softer version of "Bind markers to input" could still exist, but instead it would choose the default type markers are created as. (There's a bit more to the internals: two marker lists might be best to simplify insert/delete operations. At the very least, marker serialization into the `.tasproj` would need to change)

For visual distinction, one option is for each type of marker could have a different color by default or at least make that configurable?

Having per-marker configuration could also open the door to things like #3892

**Comment by vadosnaprimer** [comment]
For independent "markers" I just use branches. They never move, also you can backup to them.

**Comment by Morilli** [comment]
I personally tend to have the `Bind markers to input` off by default, and only when I optimize some earlier part I use it to automatically shift multiple markers at once: Activate it, insert or delete some frames, deactivate it, undo the insertion/deletion.

Is that the main use case for you also? I wonder if adding an option to shift multiple markers by a certain amount would be the most useful way to improve functionality here.

**Comment by RetroEdit** [comment]
Nah, input-bound markers is my default, because I do actively use insert/delete when TASing, so it ensure my markers stay correctly attached to their associated presses (e.g., a press to enter level 1, level 2, etc.)

Strictly speaking, feos is probably correct that branches work well enough for the common use-case that I specifically described here. I could nitpick and say branches are heavier and that they introduce branch clutter, but they can always be deleted or relegated to a backup `.tasproj`.

Fundamentally, I was a little spontaneous in filing this issue, since I have a lot more design notes and thoughts about how markers could work in the future. The main thing that motivated filing an issue was realizing that it's sort of weird and user error-prone that bind markers to input is a global TAStudio setting, when the marker schema can be pretty specific to a particular `.tasproj` file. Merely making it project-local doesn't really address the issue either because it's useful to be able to set a global default that will be applied when new project are created.

**Comment by SuuperW** [comment]
I have an idea for how to handle the user interface. We could add a new column to the marker list that indicates if it is bound to inputs. The user can toggle it by painting in the list view, just like editing inputs. We'd have an option for the default, and beyond that I think using the list would work well enough and would not add much complexity to the user interface. We can give the user the option to toggle this column's visibility, or just to re-order the columns and put it last so that it doesn't take space from the marker name.

I do not know what would be a good option for handling per-branch markers. Sometimes I want my markers to be per-branch, sometimes I don't. And this doesn't always line up with whether I want my markers to be input-bound.

Another issue that this could introduce would be having multiple markers on the same frame. Currently the user interface won't let you do that, and Lua functions assume there cannot be more than one marker per frame.
